src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -269,39 +266,7 @@ private function getDocBlockFromMethod($class, $ucFirstProperty, $type)
             return;
         }
 
-        $reflectionClass = $reflectionMethod->getDeclaringClass();
-        $fileReflector = $this->getFileReflector($reflectionClass);
-
-        if (!$fileReflector) {
-            return;
-        }
-
-        foreach ($fileReflector->getClasses() as $classReflector) {
-            $className = $this->getClassName($classReflector);
-
-            if ($className === $reflectionClass->name) {
-                if ($methodReflector = $classReflector->getMethod($methodName)) {
-                    return array($methodReflector->getDocBlock(), $prefix);
-                }
-            }
-        }
-    }
-
-    /**
-     * Gets the normalized class name (without trailing backslash).
-     *
-     * @param ClassReflector $classReflector
-     *
-     * @return string
-     */
-    private function getClassName(ClassReflector $classReflector)
-    {
-        $className = $classReflector->getName();
-        if ('\\' === $className[0]) {
-            return substr($className, 1);
-        }
-
-        return $className;
+        return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
```

Comment:
Phpdoc is wrong IMO, as it will always return an array with first argument being a Docblock and the second one being a string or null, don't know if there is a way to document that kind of return ?